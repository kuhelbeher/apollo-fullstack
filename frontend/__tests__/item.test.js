import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Item from '../components/Item';

const fakeItem = {
  id: '123',
  title: 'A Cool Item',
  price: 5000,
  description: 'This item is really cool',
  image: 'image.jpg',
  largeImage: 'largeImage.jpg',
};

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders and displays properly', () => {
    const wrapper = shallow(<Item item={fakeItem} />);

    const priceTag = wrapper.find('PriceTag');

    expect(priceTag.text()).toEqual('$50');
    expect(wrapper.find('Title a').text()).toEqual(fakeItem.title);

    const img = wrapper.find('img');

    expect(img.props().src).toEqual(fakeItem.image);
    expect(img.props().alt).toEqual(fakeItem.title);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<Item item={fakeItem} />);

    const buttonList = wrapper.find('.buttonList');

    expect(buttonList.children()).toHaveLength(3);

    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
});
