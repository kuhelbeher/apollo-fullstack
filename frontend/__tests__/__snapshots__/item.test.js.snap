// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="image.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123",
          },
        }
      }
      prefetch={true}
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item is really cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123",
          },
        }
      }
      prefetch={true}
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="123"
    />
    <DeleteItem
      id="123"
    >
      Delete
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
